lib/rules/prefer-regex-literals.js

```diff
@@ -306,10 +386,70 @@ module.exports = {
 
                 for (const { node } of tracker.iterateGlobalReferences(traceMap)) {
                     if (disallowRedundantWrapping && isUnnecessarilyWrappedRegexLiteral(node)) {
+                        const regexNode = node.arguments[0];
+
                         if (node.arguments.length === 2) {
-                            context.report({ node, messageId: "unexpectedRedundantRegExpWithFlags" });
+                            const outputRegExps = [];
+
+                            const argFlags = getStringValue(node.arguments[1]) || "";
+
+                            if (canFixTo(node, regexNode.regex.pattern, argFlags)) {
+                                outputRegExps.push({
+                                    pattern: regexNode.regex.pattern,
+                                    flags: argFlags
+                                });
+                            }
+
+                            const literalFlags = regexNode.regex.flags || "";
+
+                            if (argFlags && literalFlags) {
```

Comment:
When `argsFlags` is an empty string, for example:

```js
new RegExp(/a/u, "");
```

Currently, there's only one suggestion: `/a/`. I think there should be two suggestions: `/a/` (the equivalent one) and `/a/u` (possible intention).